01-data-structures/07-graphs/find_kevin.rb

```diff
@@ -0,0 +1,145 @@
+John_Goodman = Node.new("John Goodman")
+Wayne_Knight = Node.new("Wayne Knight")
+Kevin_Costner = Node.new("Kevin Costner")
+Lindsay_Lohan = Node.new("Lindsay Lohan")
+Natasha_Richardson = Node.new("Natasha Richardson")
+Rachel_McAdams = Node.new("Rachel McAdams")
+Amanda_Seyfried = Node.new("Amanda Seyfried")
+Ryan_Gosling = Node.new("Ryan Gosling")
+James_Garner = Node.new("James Garner")
+Sarah_Jessica_Parker = Node.new("Sarah Jessica Parker")
+
+
+Dianne_Wiest.film_actor_hash["Edward Scissorhands"] = [Johnny_Depp, Winona_Ryder]
+Dianne_Wiest.film_actor_hash["Footloose"] = [Kevin_Bacon, Chris_Penn, Sarah_Jessica_Parker]
+
+Chris_Penn.film_actor_hash["Footloose"] = [Dianne_Wiest, Kevin_Bacon, Sarah_Jessica_Parker]
+
+Sarah_Jessica_Parker.film_actor_hash["Footloose"] = [Dianne_Wiest, Kevin_Bacon, Chris_Penn]
+
+Johnny_Depp.film_actor_hash["Charlie and the Chocolate Factory"] = [Freddie_Highmore, AnnaSophia_Robb]
+Johnny_Depp.film_actor_hash["Edward Scissorhands"] = [Dianne_Wiest, Winona_Ryder]
+
+Winona_Ryder.film_actor_hash["Edward Scissorhands"] = [Johnny_Depp, Dianne_Wiest]
+
+Freddie_Highmore.film_actor_hash["Charlie and the Chocolate Factory"] = [Johnny_Depp, AnnaSophia_Robb]
+
+AnnaSophia_Robb.film_actor_hash["Charlie and the Chocolate Factory"] = [Johnny_Depp, Freddie_Highmore]
+AnnaSophia_Robb.film_actor_hash["Soul Surfer"] = [Carrie_Underwood, Dennis_Quaid]
+
+Carrie_Underwood.film_actor_hash["Soul Surfer"] = [AnnaSophia_Robb, Dennis_Quaid]
+
+Dennis_Quaid.film_actor_hash["Soul Surfer"] = [AnnaSophia_Robb, Carrie_Underwood]
+Dennis_Quaid.film_actor_hash["Everybody's All-American"] = [John_Goodman, Wayne_Knight]
+Dennis_Quaid.film_actor_hash["The Parent Trap"] = [Lindsay_Lohan, Natasha_Richardson]
+
+John_Goodman.film_actor_hash["Everybody's All-American"] = [Dennis_Quaid, Wayne_Knight]
+
+Wayne_Knight.film_actor_hash["JFK"] = [Kevin_Costner, Kevin_Bacon]
+
+Kevin_Costner.film_actor_hash["JFK"] = [Wayne_Knight, Kevin_Bacon]
+
+Lindsay_Lohan.film_actor_hash["The Parent Trap"] = [Dennis_Quaid, Natasha_Richardson]
+Lindsay_Lohan.film_actor_hash["Mean Girls"] = [Rachel_McAdams, Amanda_Seyfried]
+
+Rachel_McAdams.film_actor_hash["Mean Girls"] = [Amanda_Seyfried, Lindsay_Lohan]
+Rachel_McAdams.film_actor_hash["The Notebook"] = [Ryan_Gosling, James_Garner]
+
+Ryan_Gosling.film_actor_hash["The Notebook"] = [James_Garner, Rachel_McAdams]
+
+James_Garner.film_actor_hash["The Notebook"] = [Rachel_McAdams, Ryan_Gosling]
+
+Amanda_Seyfried.film_actor_hash["Mean Girls"] = [Lindsay_Lohan, Rachel_McAdams]
+
+Natasha_Richardson.film_actor_hash["The Parent Trap"] = [Lindsay_Lohan, Dennis_Quaid]
+
+Wayne_Knight.film_actor_hash["Everybody's All-American"] = [John_Goodman, Dennis_Quaid]
+
+# all_paths = Array.new() # sorted, shortest path at 0, longest at n-1
+# start = Freddie_Highmore
+# path, path_weight = find_kevin_bacon(start)
+# all_paths.push(path)
+# shortest_path = all_paths[0]
+
+$path = []
+$paths = []
+
+def included_film?(film)
```

Comment:
- [ ] You should add comments here are the top describing what this method does as well as the parameters it takes and what it returns.